### PR TITLE
Upgrade Rubocop to 1.15

### PR DESCRIPTION
**Pull request by teceler**
Depends on #1518

- Upgrades rubocop to 1.15.0
- Upgrades rubocop-performance to 1.11.3
- Upgrades rubocop-rails to 2.10.1
- Upgrades rubocop-rspec to 2.3.0
&nbsp;

**New (enabled) cops:**
- Gemspec/DateAssignment
- ~~Layout/SpaceBeforeBrackets~~
- Lint/AmbiguousAssignment
- Lint/DeprecatedConstants
- Lint/DuplicateBranch
- Lint/DuplicateRegexpCharacterClassElement
- Lint/EmptyBlock
- Lint/EmptyClass
- Lint/LambdaWithoutLiteralBlock
- Lint/NoReturnInBeginEndBlocks
- Lint/NumberedParameterAssignment
- Lint/OrAssignmentToConstant
- Lint/RedundantDirGlobSort
- Lint/SymbolConversion
- Lint/ToEnumArguments
- Lint/TripleQuotes
- Lint/UnexpectedBlockArity
- Lint/UnmodifiedReduceAccumulator
- Style/ArgumentsForwarding
- Style/CollectionCompact
- Style/DocumentDynamicEvalDefinition
- Style/EndlessMethod
- Style/HashConversion
- Style/HashExcept
- Style/IfWithBooleanLiteralBranches
- Style/NegatedIfElseCondition
- Style/NilLambda
- ~~Style/RedundantArgument~~
- Style/StringChars
- Style/SwapValues
- Performance/MapCompact

### CODE REVIEW

File: spec/models/favorite_spec.rb [hunk 1]
```diff
@@ -43,7 +43,6 @@
       expect(f2.save).to eq(true)
     end
 
-    it "should allow you to favorite multiple things" do
-    end
+    it "should allow you to favorite multiple things"
```
Comment (Throne3d) [inline spec/models/favorite_spec.rb:46]:
```suggestion
    skip "should allow you to favorite multiple things"
```
?

File: .rubocop.yml [hunk 2]
```diff
@@ -83,15 +85,15 @@ Layout/ParameterAlignment:
...
-Layout/SpaceAroundMethodCallOperator:
-  Enabled: true
-
 Layout/SpaceAroundOperators:
   Enabled: false
 
 Layout/SpaceBeforeBlockBraces:
   Enabled: false
 
+Layout/SpaceBeforeBrackets: # (new in 1.7)
```
Comment (Throne3d) [inline .rubocop.yml:94]: This cop seems to have issues, especially with situations like the example you found in our code – https://github.com/rubocop/rubocop/issues/9438, https://github.com/rubocop/rubocop/issues/9290

Maybe we should disable it for now?

Comment (teceler) [reply .rubocop.yml:94]: ...wow that's impressive. I can do that, yes.

File: app/concerns/authentication/api.rb [hunk 3]
```diff
@@ -26,7 +26,7 @@ def decoded_api_token
 
     # handle edge case errors more reasonably
     def decode_api_token
-      auth_header = request.headers['Authorization'].to_s.split(' ').last
+      auth_header = request.headers['Authorization'].to_s.split.last
```
Comment (Throne3d) [inline app/concerns/authentication/api.rb:29]: Which lint affects this? I don't like this change much – it looks like [there's some fallback to a `$;` global when this isn't specified](https://ruby-doc.org/core-3.0.1/String.html#method-i-split), and it seems a little magic to rely on this being a single whitespace, but maybe I'm just being a bit weird :P

**Review by Throne3d — APPROVED** [review]
Looks OK beyond the comments! We can always tweak the lints as we go, if they get in our way – I haven't checked them out in much detail

File: app/concerns/authentication/api.rb (hunk 3 above)
Comment (teceler) [reply app/concerns/authentication/api.rb:29]: [Style/RedundantArgument](https://docs.rubocop.org/rubocop/cops_style.html#styleredundantargument), apparently

**Pull request merged**